### DISCUSSION TITLE
Conda environment file for macOS

### DISCUSSION THREAD

**Discussion by roopchansinghv** [gadgetron/gadgetron#1141] [General]
As I've been getting Gadgetron building on macOS, there are different requirements for macOS on Conda, versus Linux.

The most prominent ones are the swapping of Clang for gcc, as the version of gcc available for macOS on anaconda.org is 4.8.5 versus 12.1.0 for all other platforms, and the non-existence of CUDA support libraries for macOS.

I tried if the platform-specific flags used in the gadgetron-conda-recipes repository (like [here](https://github.com/gadgetron/gadgetron-conda-recipes/blob/master/ismrmrd/meta.yaml)) would work in an environment creation file, but they don't seem to.

Are we okay with just adding an environment-macos.yml file right along side Gadgetron's default environment.yml file?  If not - what are alternatives?

**Comment by hansenms** [comment]
I think we have some different options, but it would be critical to me that this environment file is somehow tested. Do you have anything creative ideas on this?

**Comment by roopchansinghv** [comment]
Yes - I think you should be able to continue sending the Linux builds and testing to Azure, and we can leverage something like [this](https://github.com/gadgetron/gadgetron-conda-recipes/blob/0851cf2e4c2478bdef32bafb4870c8451844ba26/.github/workflows/build.yml#L40) for the macOS builds, using GitHub's built in workflows.

**Comment by hansenms** [comment]
Do you think we could add it to the Azure Devops pipeline (https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/hosted?view=azure-devops&tabs=yaml) to not introduce two different build systems?

**Comment by roopchansinghv** [comment]
Absolutely - if MS is okay with us using the macOS machines for this, I am more than happy to use that system in parallel.  Hopefully having a reduced set of tests for macOS helps.  Let me take a stab at this and see how far I get.

**Comment by hansenms** [comment]
Yeah, lets go with that.
